deltacat/compute/compactor_v2/utils/compaction_utils.py

```diff
@@ -0,0 +1,764 @@
+
+    delete_strategy: Optional[DeleteStrategy] = None
+    delete_file_envelopes: Optional[List[DeleteFileEnvelope]] = None
+    delete_file_size_bytes: int = 0
+    if contains_delete_deltas(input_deltas):
+        input_deltas, delete_file_envelopes, delete_strategy = prepare_deletes(
+            params, input_deltas
+        )
+        for delete_file_envelope in delete_file_envelopes:
+            delete_file_size_bytes += delete_file_envelope.table_size_bytes
+        logger.info(
+            f" Input deltas contain {DeltaType.DELETE}-type deltas. Total delete file size={delete_file_size_bytes}."
+            f" Total length of delete file envelopes={len(delete_file_envelopes)}"
+        )
+    uniform_deltas: List[DeltaAnnotated] = io.create_uniform_input_deltas(
+        input_deltas=input_deltas,
+        hash_bucket_count=params.hash_bucket_count,
+        compaction_audit=mutable_compaction_audit,
+        deltacat_storage=params.deltacat_storage,
+        previous_inflation=params.previous_inflation,
+        min_delta_bytes=params.min_delta_bytes_in_batch,
+        min_file_counts=params.min_files_in_batch,
+        # disable input split during rebase as the rebase files are already uniform
+        enable_input_split=params.rebase_source_partition_locator is None,
+        deltacat_storage_kwargs=params.deltacat_storage_kwargs,
+    )
+    delta_discovery_end: float = time.monotonic()
+
+    mutable_compaction_audit.set_uniform_deltas_created(len(uniform_deltas))
+    mutable_compaction_audit.set_delta_discovery_time_in_seconds(
+        delta_discovery_end - delta_discovery_start
+    )
+
+    s3_utils.upload(
+        mutable_compaction_audit.audit_url,
+        str(json.dumps(mutable_compaction_audit)),
+        **params.s3_client_kwargs,
+    )
+
+    # create a new stream for this round
+    compacted_stream_locator: Optional[
+        StreamLocator
+    ] = params.destination_partition_locator.stream_locator
+    compacted_stream: Stream = params.deltacat_storage.get_stream(
+        compacted_stream_locator.namespace,
+        compacted_stream_locator.table_name,
+        compacted_stream_locator.table_version,
+        **params.deltacat_storage_kwargs,
+    )
+    compacted_partition: Partition = params.deltacat_storage.stage_partition(
+        compacted_stream,
+        params.destination_partition_locator.partition_values,
+        **params.deltacat_storage_kwargs,
+    )
+    return (
+        uniform_deltas,
+        delete_strategy,
+        delete_file_envelopes,
+        compacted_partition,
+    )
+
+
+def _run_merge(
+    params: CompactPartitionParams,
+    uniform_deltas,
+    compacted_partition,
+    round_completion_info,
+    delete_strategy,
+    delete_file_envelopes,
+    mutable_compaction_audit,
+    previous_compacted_delta_manifest,
+    task_max_parallelism,
+) -> tuple[list[MergeResult], np.int64, Any, Any, Any, Any]:
```

Comment:
I think the Any's are `float`, `int`, `int`, `float`.

Do you think it might be possible to simplify the output of `_run_merge` and correspondingly the input args of `_complete_round`?

Also, perhaps you could use a more descriptive name for `_complete_round`. 

Anywhere else you can think of ways to simplify the inputs and outputs of these compaction utils, that'd help.